arch/arm/src/armv7-m/arm_cache.c

```diff
@@ -529,7 +529,7 @@ void up_enable_dcache(void)
   ARM_DSB();
   do
     {
-      int32_t tmpways = ways;
+      uint32_t tmpways = ways;
```

Comment:
but no really difference I think?

Reply:
do you really get a negative number from real device? This register is read only and officially defined by arm:
https://developer.arm.com/documentation/ddi0595/2021-03/AArch32-Registers/CCSIDR--Current-Cache-Size-ID-Register.
`CCSIDR_WAYS(ccsidr)` expends to `(ccsider >> 3) & 0x3ff`, it is impossible to become a negative number.